packages/discord.js/src/structures/MessagePayload.js

```diff
@@ -131,9 +132,7 @@ class MessagePayload {
       }
     }
 
-    const components = this.options.components?.map(c =>
-      isJSONEncodable(c) ? c.toJSON() : this.target.client.options.jsonTransformer(c),
-    );
+    const components = this.options.components?.map(c => (isJSONEncodable(c) ? c : new ActionRowBuilder(c)).toJSON());
```

Comment:
This is incorrect, `JSONEncodable` means that the object has a `toJSON` method it doesn't mean the object is JSON itself.

```suggestion
    const components = this.options.components?.map(c => (isJSONEncodable(c) ? c.toJSON() : new ActionRowBuilder(c)).toJSON());
```

Reply:
you should probably read the full thing, this is still calling toJSON on c

Reply:
Ah I see. However I noticed another problem, `ActionRowBuilder` *is* JSONEncodable so only one branch of the condition will ever execute.

Reply:
what? This is simply creating an action row builder if there isn't one already aka if it isn't a builder instance

Reply:
Yeah, I don't know what I'm on right now. Ignore me